Add event detector

FedericoCeratto opened the pull request:
Add event detector and RSS feed generation
Implements #229
Related to #213 and https://github.com/ooni/spec/issues/144

Commit:
Add event detector

bassosimone commented on af/fastpath/debian/detector.service:
```diff
@@ -0,0 +1,39 @@
+ProtectHome=yes
+#ProtectHome=read-only
+ProtectSystem=strict
+ProtectSystem=full
```
Is `ProtectSystem=full` overriding or augmenting `ProtectSystem=strict`?

FedericoCeratto replied:
It's overriding it, I should remove the other.
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Sandboxing

bassosimone commented on af/fastpath/fastpath/detector.py:
```diff
@@ -0,0 +1,520 @@
+    AND input = %(inp)s
+
+    ORDER BY measurement_start_time
+    """
```
I assume your end goal is to use this function to implement also the one above?

FedericoCeratto replied:
Those are 2 different use cases: filtering by test_name/cc/input or getting everything.

bassosimone commented on af/fastpath/debian/control:
```diff
@@ -14,6 +14,7 @@ Depends: ${misc:Depends},
  python3-statsd,
- python3-systemd
+ python3-systemd,
+ nginx
```
Is this redundant, considering that we are using Ansible to deploy hosts?

FedericoCeratto replied:
The dependencies are also used to run devel version on workstations,  run unit/integ tests and ensure that we are really deploying the required deps at runtime

bassosimone reviewed (commented):
Hello @FedericoCeratto, here are my preliminary comments!

I like the principle of using Debian functionality as much as possible! I am unfortunately not familiar enough yet with our Ansible infrastructure, so I've asked a question regarding that.

I believe it would be beneficial to factor the bits where we perform measurement analysis in a different module with unit tests, so it would be easier (at least for me) to understand and visualize what is the typical input and the typical output. (That is, I'm not advocating for full coverage, even though this is nice, but rather to have unit tests as live examples of building blocks functionality.)

Thank you!

hellais commented:
I am a bit concerned of the approach being so much reliant on debian packages for deploying stuff.

I fear this may lead to bits of our infrastructure using a certain pattern and others using a different one, making it necessary to know more bits of the system in order to do a new deployment.

I guess that if you publish the debian packages to some location and then the ansible playbook fetches a prebuilt packages, that is ok, though I would like to understand better what is the reason to go away from the current ansible based infrastructure.

I think it's easier to reason about a system (and make cross-cutting changes) when the patterns used in all the infrastructure are few and consistent.

I don't think it's necessary to use Docker (though it does make our life significantly easier in many regards), but I would like to better understand what direction you are thinking of going by introducing debian packaging in here.

Commit:
Add very basic unit test

Commit:
Implement file output, add Nginx configuration

Commit:
Detector webapp: split by AS

Commit:
Improve logging, add comments

Commit:
Minor changes

Commit:
Scale up charts

Commit:
Handle future dates

Commit:
Update moving avg params

Commit:
Finetune detection params, align charts

Improve chart colors

Commit:
Log tid/report_id from future

Commit:
Add RO database option

Commit:
Show multiple charts

Commit:
add global and by-CC feeds

Commit:
Move detector into an independent component

Commit:
Snapshot means every month to allow reprocessing

Commit:
Add metrics

Commit:
Cleanup, skip unused code, add metric

Commit:
Add country name support

Commit:
Detector 0.2 - run webapp

Commit:
Add metrics file

Commit:
Add scoring file

Commit:
Fix scoring file

Commit:
Add d/s/format

Commit:
Add __init__.py file

Commit:
Update setup.py

Commit:
Use /etc/detector.conf for DB hostnames

Allow easier ansible deployment and manual override on testbeds and
workstations

hellais reviewed (approved):
We further discussed the deb based deployed and I think we have found a solution which can work.

For me this branch is good to be landed in master.
